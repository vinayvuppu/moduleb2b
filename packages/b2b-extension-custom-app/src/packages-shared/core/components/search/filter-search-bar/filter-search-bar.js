import PropTypes from 'prop-types';
import React from 'react';
import { defineMessages, injectIntl } from 'react-intl';
import {
  CloseBoldIcon,
  FilterIcon,
  FlatButton,
  SecondaryButton,
} from '@commercetools-frontend/ui-kit';
import SearchInput from '../../fields/search-input';
import styles from './filter-search-bar.mod.css';

const messages = defineMessages({
  filterLabel: {
    id: 'Search.FilterSearchBar.filterLabel',
    description: 'Label for "Filter" button.',
    defaultMessage: 'Filter',
  },
  removeAllLabel: {
    id: 'Search.FilterSearchBar.clearAllLabel',
    description: 'Label for the button for clear all',
    defaultMessage: 'Clear all',
  },
});

export const FilterSearchBar = props => (
  <div>
    <div className={styles.container}>
      <div className={styles['toggle-filter-button']}>
        <SecondaryButton
          label={props.intl.formatMessage(messages.filterLabel)}
          isDisabled={props.isFilterButtonDisabled}
          onClick={props.onToggleFilterButton}
          isToggled={props.isFilterButtonActive}
          isToggleButton={true}
          iconLeft={<FilterIcon />}
          data-track-component=""
          data-track-label="Filters"
          data-track-event="click"
        />
      </div>
      <SearchInput
        initialValue={props.initialValue}
        onSubmit={props.onSubmit}
        onChange={props.onChange}
        throttleMS={
          1 /* we pass 1ms to prevent user search before state update */
        }
        placeholder={props.searchInputPlaceholder}
      />
      <div className={styles['clear-button']}>
        {props.showOnClearAll && (
          <FlatButton
            data-testid="clear-all-filters"
            onClick={props.onClearAll}
            icon={<CloseBoldIcon />}
            tone="secondary"
            label={props.intl.formatMessage(messages.removeAllLabel)}
          />
        )}
      </div>
    </div>
    {props.renderQuickFilters({
      onUpdateQuickFilterForField: props.onUpdateQuickFilterForField,
      onRemoveFilterTagFromField: props.onRemoveFilterTagFromField,
    })}
  </div>
);
FilterSearchBar.displayName = 'FilterSearchBar';
FilterSearchBar.propTypes = {
  intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired })
    .isRequired,
  initialValue: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  onToggleFilterButton: PropTypes.func.isRequired,
  isFilterButtonActive: PropTypes.bool.isRequired,
  isFilterButtonDisabled: PropTypes.bool.isRequired,
  onChange: PropTypes.func,
  onClearAll: PropTypes.func.isRequired,
  showOnClearAll: PropTypes.bool.isRequired,
  searchInputPlaceholder: PropTypes.string,

  renderQuickFilters: PropTypes.func,
  onUpdateQuickFilterForField: PropTypes.func.isRequired,
  onRemoveFilterTagFromField: PropTypes.func.isRequired,
};
FilterSearchBar.defaultProps = {
  renderQuickFilters: () => null,
};

export default injectIntl(FilterSearchBar);
