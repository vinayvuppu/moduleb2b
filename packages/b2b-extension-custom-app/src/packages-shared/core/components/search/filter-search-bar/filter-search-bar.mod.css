.container {
  display: inline-flex;
  width: 100%;
  padding: var(--spacing-s);
}

.toggle-filter-button {
  margin: 0 var(--spacing-s) 0 0;
}

.clear-button {
  flex: 1;
  display: flex;
  justify-content: flex-end;
  align-self: flex-end;
}

.clear-button i {
  color: var(--color-solid);
}
