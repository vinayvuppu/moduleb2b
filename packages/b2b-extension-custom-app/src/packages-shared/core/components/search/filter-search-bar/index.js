export { default } from './filter-search-bar';
