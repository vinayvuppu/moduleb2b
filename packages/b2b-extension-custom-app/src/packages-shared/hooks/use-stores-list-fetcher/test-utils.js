import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { FetchStoresQuery } from './stores-list-query.graphql';

// eslint-disable-next-line import/prefer-default-export
export const createFetchStoresListQueryMock = (custom = {}) => ({
  request: {
    variables: {
      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
      limit: 20,
      offset: 0,
      sort: `createdAt desc`,
    },
    ...custom.request,
    query: FetchStoresQuery,
  },
  result: {
    data: {
      stores: {
        count: 2,
        offset: 0,
        total: 2,
        results: [
          {
            id: 'store-1',
            key: 'germany',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
            createdAt: '2018-12-19T15:26:51.174Z',
            lastModifiedAt: '2019-03-01T09:00:51.174Z',
          },
          {
            id: 'store-2',
            key: 'usa',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
            createdAt: '2019-06-10T15:00:51.140Z',
            lastModifiedAt: '2019-08-30T10:26:51.140Z',
          },
        ],
      },
    },
    ...custom.result,
  },
});
