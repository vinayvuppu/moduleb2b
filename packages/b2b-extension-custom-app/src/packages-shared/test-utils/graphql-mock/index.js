export { createRosie, createGraphqlMock } from './graphql-mock';
