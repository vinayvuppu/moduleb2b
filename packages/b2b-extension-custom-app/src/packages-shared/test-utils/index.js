import crypto from 'crypto';
import { withProps, compose, branch, renderComponent } from 'recompose';

export { createGraphqlMock, createRosie } from './graphql-mock';

export const intlMock = {
  formatMessage: jest.fn(message => message.id),
  formatDate: jest.fn(date => {
    const dateObj = new Date(date);
    const day = dateObj.getUTCDate();
    const month = dateObj.getUTCMonth() + 1;
    const year = dateObj.getUTCFullYear();
    return `${day}.${month}.${year}`;
  }),
  formatTime: jest.fn(date => {
    const dateObj = new Date(date);
    const hours = dateObj.getUTCHours();
    const minutes = dateObj.getUTCMinutes();
    return `${hours}:${minutes}`;
  }),
  formatRelative: jest.fn(() => 'xxx'),
  formatNumber: jest.fn((number, currency) =>
    currency?.currency
      ? `${currency.currency} ${number.toString()}`
      : number.toString()
  ),
  formatPlural: jest.fn(() => 'xxx'),
  formatHTMLMessage: jest.fn(() => 'xxx'),
  now: jest.fn(() => 'xxx'),
  locale: 'en',
};

export const routerMock = {
  go: jest.fn(() => {}),
  goBack: jest.fn(() => {}),
  goForward: jest.fn(() => {}),
  push: jest.fn(() => {}),
  replace: jest.fn(() => {}),
  isActive: jest.fn(() => {}),
  createHref: jest.fn(() => {}),
  setRouteLeaveHook: jest.fn(() => {}),
};

export function createMountOptions(options = {}) {
  const { intl, router } = options;
  return {
    context: {
      intl: { ...intlMock, ...intl },
      router: { ...routerMock, ...router },
      store: {
        getState: () => ({
          application: {
            products: {
              currentVisible: {
                productType: {
                  obj: {
                    name: 'foo',
                  },
                },
              },
            },
          },
        }),
        subscribe() {},
        dispatch() {},
      },
    },
  };
}

export function promisedTimeout(cb, timeout = 0) {
  const p = new Promise(resolve => {
    setTimeout(() => {
      resolve(cb());
    }, timeout);
  });
  return p;
}

export const randomId = () => crypto.randomBytes(16).toString('hex');

export const mockPermissions = ({ isAuthorized = true } = {}) => {
  const injectAuthorized = (
    demandedPermissions,
    options,
    propName = 'isAuthorized'
  ) => withProps({ [propName]: isAuthorized });

  const branchOnPermissions = (
    demandedPermissions,
    FallbackComponent,
    options
  ) =>
    compose(
      injectAuthorized(demandedPermissions, options),
      branch(() => !isAuthorized, renderComponent(FallbackComponent))
    );
  return {
    injectAuthorized,
    branchOnPermissions,
    permissions: {},
  };
};

export const mockBranchOnFeatureToggle = ({
  isFeatureToggledOn = true,
} = {}) => (flags, FallbackComponent) =>
  branch(() => !isFeatureToggledOn, renderComponent(FallbackComponent));

export const mockComponent = displayName => {
  const MockedComponent = ({ children }) => children || null;
  MockedComponent.displayName = displayName;
  return MockedComponent;
};

export const mockConnector = (displayName, childrenProps = {}) => {
  const MockedConnector = jest.fn(({ children }) => children(childrenProps));
  MockedConnector.displayName = displayName;
  return MockedConnector;
};

export const mockContext = displayName => {
  let providedValue;

  const ProviderComponent = ({ value, children }) => {
    providedValue = value;
    return children;
  };
  ProviderComponent.displayName = `${displayName}.Provider`;
  const ConsumerComponent = ({ children }) => children(providedValue);
  ConsumerComponent.displayName = `${displayName}.Consumer`;

  return {
    Provider: ProviderComponent,
    Consumer: ConsumerComponent,
  };
};

export const mockReactIntl = () => ({
  ...require.requireActual('react-intl'),
  FormattedMessage: ({ id }) => `formatted-${id}`,
  injectIntl: withProps({ intl: intlMock }),
});

export const waitForPromises = () =>
  new Promise(resolve => setImmediate(resolve));
