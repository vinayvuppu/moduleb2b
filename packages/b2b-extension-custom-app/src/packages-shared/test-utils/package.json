{
  "name": "@commercetools-local/test-utils",
  "version": "1.0.0",
  "description": "",
  "license": "Proprietary",
  "private": true,
  "main": "./index.js",
  "dependencies": {
    "@commercetools-frontend/constants": "15.3.4",
    "@commercetools-frontend/notifications": "15.3.4",
    "@commercetools-local/utils": "*",
    "faker": "4.1.0",
    "graphql": "14.5.8",
    "graphql-tools": "4.0.4",
    "lodash.merge": "4.6.2",
    "prop-types": "15.7.2",
    "recompose": "0.26.0",
    "rosie": "2.0.1"
  }
}
