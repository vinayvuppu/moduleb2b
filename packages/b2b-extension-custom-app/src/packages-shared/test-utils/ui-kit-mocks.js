/* eslint-disable global-require, react/prop-types, react/display-name */
jest.mock('@commercetools-frontend/ui-kit', () => {
  const React = require('react');
  class MockedTable extends React.Component {
    componentDidMount() {
      if (this.props.registerMeasurementCache) {
        this.props.registerMeasurementCache({
          clearAll: () => {},
        });
      }
    }
    render() {
      return (
        <div>
          <table>
            <thead>
              <tr>
                {this.props.columns.map(column => (
                  <td key={column.key}>{column.label}</td>
                ))}
              </tr>
            </thead>
            <tbody>
              {this.props.items.map((item, rowIndex) => (
                <tr key={rowIndex}>
                  {this.props.columns.map(column => (
                    <td
                      key={column.key}
                      data-testid={`cell-${rowIndex}-${column.key}`}
                      onClick={() =>
                        this.props.onRowClick
                          ? this.props.onRowClick(column.key, rowIndex)
                          : null
                      }
                    >
                      {this.props.itemRenderer({
                        rowIndex,
                        columnKey: column.key,
                      })}
                    </td>
                  ))}
                </tr>
              ))}
            </tbody>
          </table>
          {this.props.children}
        </div>
      );
    }
  }
  return {
    ...jest.requireActual('@commercetools-frontend/ui-kit'),
    Table: MockedTable,
  };
});
