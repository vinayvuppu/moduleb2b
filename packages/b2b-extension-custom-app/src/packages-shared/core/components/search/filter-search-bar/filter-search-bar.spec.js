import React from 'react';
import { shallow } from 'enzyme';
import { FlatButton } from '@commercetools-frontend/ui-kit';
import { intlMock } from '../../../../test-utils';
import { FilterSearchBar } from './filter-search-bar';

const createTestProps = props => ({
  intl: intlMock,
  initialValue: '123',
  onSubmit: jest.fn(),
  onToggleFilterButton: jest.fn(),
  isFilterButtonActive: false,
  isFilterButtonDisabled: false,
  onChange: jest.fn(),
  onClearAll: jest.fn(),
  showOnClearAll: false,
  searchInputPlaceholder: 'Some search placeholder...',
  onUpdateQuickFilterForField: jest.fn(),
  onRemoveFilterTagFromField: jest.fn(),
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<FilterSearchBar {...props} />);
  });

  it('outputs correct tree', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('outputs with filter button disabled', () => {
    wrapper.setProps({ isFilterButtonDisabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  describe('rendering when exists text on SearchInput', () => {
    beforeEach(() => {
      props = createTestProps({ showOnClearAll: true });
      wrapper = shallow(<FilterSearchBar {...props} />);
    });
    it('should render the `remove All` button', () => {
      expect(wrapper).toRender(FlatButton);
    });
  });

  describe('when there are quick filters', () => {
    beforeEach(() => {
      props = createTestProps({
        renderQuickFilters: jest.fn().mockName('renderQuickFilters'),
      });
      wrapper = shallow(<FilterSearchBar {...props} />);
    });
    it('should invoke `renderQuickFilters`', () => {
      expect(props.renderQuickFilters).toHaveBeenCalled();
    });
    it('should invoke `renderQuickFilters` with `onUpdateQuickFilterForField`', () => {
      expect(props.renderQuickFilters).toHaveBeenCalledWith(
        expect.objectContaining({
          onUpdateQuickFilterForField: props.onUpdateQuickFilterForField,
        })
      );
    });
    it('should invoke `renderQuickFilters` with `onRemoveFilterTagFromField`', () => {
      expect(props.renderQuickFilters).toHaveBeenCalledWith(
        expect.objectContaining({
          onRemoveFilterTagFromField: props.onRemoveFilterTagFromField,
        })
      );
    });
  });
});

describe('callbacks', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps({ showOnClearAll: true });
    wrapper = shallow(<FilterSearchBar {...props} />);
    wrapper.find(FlatButton).simulate('click');
  });
  it('should call on `oncClearAll` function', () => {
    expect(props.onClearAll).toHaveBeenCalledTimes(1);
  });
});
