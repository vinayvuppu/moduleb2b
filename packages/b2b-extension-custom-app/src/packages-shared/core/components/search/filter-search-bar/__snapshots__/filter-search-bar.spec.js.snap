// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="toggle-filter-button"
    >
      <SecondaryButton
        data-track-component=""
        data-track-event="click"
        data-track-label="Filters"
        iconLeft={<ForwardRef(render) />}
        isDisabled={false}
        isToggleButton={true}
        isToggled={false}
        label="Search.FilterSearchBar.filterLabel"
        onClick={[MockFunction]}
        theme="default"
        type="button"
      />
    </div>
    <SearchInput
      initialValue="123"
      onChange={[MockFunction]}
      onSubmit={[MockFunction]}
      placeholder="Some search placeholder..."
      throttleMS={1}
    />
    <div
      className="clear-button"
    />
  </div>
</div>
`;

exports[`rendering outputs with filter button disabled 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="toggle-filter-button"
    >
      <SecondaryButton
        data-track-component=""
        data-track-event="click"
        data-track-label="Filters"
        iconLeft={<ForwardRef(render) />}
        isDisabled={true}
        isToggleButton={true}
        isToggled={false}
        label="Search.FilterSearchBar.filterLabel"
        onClick={[MockFunction]}
        theme="default"
        type="button"
      />
    </div>
    <SearchInput
      initialValue="123"
      onChange={[MockFunction]}
      onSubmit={[MockFunction]}
      placeholder="Some search placeholder..."
      throttleMS={1}
    />
    <div
      className="clear-button"
    />
  </div>
</div>
`;
